File: 프로그래머스/연습문제/Lv1/대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    alphas = [101 for i in range(26)]
    ordA = ord('A')
    for key in keymap:
        for i in range(len(key)):
            alphas[ord(key[i])-ordA] = min(alphas[ord(key[i])-ordA], i+1)

    answer = []
    for target in targets:
        sum = 0
        for t in target:
            if alphas[ord(t)-ordA] == 101:
                sum = -1
                break
            else: sum += alphas[ord(t)-ordA]
        answer.append(sum)

    return answer

File: 프로그래머스/연습문제/Lv1/대충 만든 자판/대충 만든 자판.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

vector<int> solution(vector<string> keymap, vector<string> targets) {
    vector<int> alphas(26, 101);
    for(string key : keymap){
        for(int i=0; i<key.length(); i++)
            alphas[key[i]-'A'] = min(alphas[key[i]-'A'], i+1);
    }
    
    vector<int> answer;
    for(string target : targets){
        int sum = 0;
        for(int i=0; i<target.length(); i++){
            if(alphas[target[i]-'A'] == 101){
                sum = -1;
                break;
            }
            else sum += alphas[target[i]-'A'];
        }
        answer.push_back(sum);
    }
    
    return answer;
}
